source/Camera/Advanced/AllocateMemoryForPointCloudData/AllocateMemoryForPointCloudData.cpp

```diff
@@ -0,0 +1,95 @@
+With the first method, the memory is allocated by user (OpenCV) based on the camera resolution before any capture
+occurs. With the second method, the memory is allocated by Zivid SDK based on the point cloud resolution, so
+post-capture.
+
+Note: This example uses experimental SDK features, which may be modified, moved, or deleted in the future without
+notice.
+*/
+
+#include <Zivid/Experimental/SettingsInfo.h>
+#include <Zivid/Zivid.h>
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <iostream>
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::cout << "Connecting to camera" << std::endl;
+        auto camera = zivid.connectCamera();
+
+        std::cout << "Creating settings" << std::endl;
+        auto settings = Zivid::Settings{ Zivid::Settings::Acquisitions{ Zivid::Settings::Acquisition{} } };
+
+        std::cout << "Getting camera resolution" << std::endl;
+        const auto resolution = Zivid::Experimental::SettingsInfo::resolution(camera.info(), settings);
+
+        // Copy selected data from GPU to system memory(user - allocated)
+        // https://support.zivid.com/latest/api/porting-guide-2_0/porting-guide-cpp.html#copy-selected-data-from-gpu-to-system-memory-user-allocated
+
+        std::cout << "OpenCV allocating the necessary storage based on resolution "
+                     "info before any capturing"
+                  << std::endl;
+        auto rgbaUA = cv::Mat(resolution.height(), resolution.width(), CV_8UC4);
+        auto *dataPtr = reinterpret_cast<Zivid::ColorRGBA *>(rgbaUA.data);
+
+        std::cout << "Capturing frame" << std::endl;
+        auto frame = camera.capture(settings);
+        auto pointCloud = frame.pointCloud();
+
+        std::cout << "Zivid API copying data from the GPU into the memory location "
+                     "allocated by OpenCV"
+                  << std::endl;
+        pointCloud.copyData(dataPtr);
+
+        std::cout << "Displaying image" << std::endl;
+        cv::Mat bgrUA;
+        cv::cvtColor(rgbaUA, bgrUA, cv::COLOR_RGBA2BGR);
+        cv::imshow("BGR image User Allocated", bgrUA);
+        cv::waitKey(0);
+
+        // Copy selected data from GPU to system memory (Zivid-allocated)
+        // https://support.zivid.com/latest/api/porting-guide-2_0/porting-guide-cpp.html#copy-selected-data-from-gpu-to-system-memory-zivid-allocated
+
+        std::cout << "Capturing frame" << std::endl;
+        frame = camera.capture(settings);
+        pointCloud = frame.pointCloud();
+
+        std::cout << "Zivid copying colors from GPU to CPU" << std::endl;
+        auto colors = pointCloud.copyColorsRGBA();
+
+        std::cout << "Casting the data pointer as a void*, since this is what the "
+                     "OpenCV matrix constructor requires."
+                  << std::endl;
+        auto *dataPtrZA = const_cast<void *>(static_cast<const void *>(colors.data()));
```

Comment:
What is ZA short for? Can we spell it out? (Same comment goes for other variables further down)

Reply:
Fied in https://github.com/zivid/zivid-cpp-samples/pull/178/commits/7d1ce97c9a9d8d17f93895fac48852af6c0a2d40